File: src/dataset/c/isogram_Exercism_attilahorvath/claude/ClaudeSonnet4_isogram_v2.c
#include "isogram.h"
#include <ctype.h>

int isIsogram(const char *string) {
    unsigned int seen = 0;
    int c;
    
    for (int i = 0; (c = string[i]) != '\0'; i++) {
        if (c >= 'A' && c <= 'Z') {
            c = c + 32;
        } else if (c < 'a' || c > 'z') {
            continue;
        }
        
        unsigned int bit = 1u << (c - 'a');
        if (seen & bit) {
            return 0;
        }
        seen |= bit;
    }
    
    return 1;
}

File: src/dataset/go/isogram_Exercism_go_ThomasZumsteg/claude/ClaudeSonnet4_isogram_v2.go
package isogram

import "unicode"

func IsIsogram(s string) bool {
	var seen uint32
	for _, char := range s {
		char = unicode.ToLower(char)
		if 'a' <= char && char <= 'z' {
			bit := uint32(1) << (char - 'a')
			if seen&bit != 0 {
				return false
			}
			seen |= bit
		}
	}
	return true
}